ecs/sdk.go

```diff
@@ -317,29 +317,113 @@ func (s sdk) ListStacks(ctx context.Context, name string) ([]compose.Stack, erro
 	}
 	stacks := []compose.Stack{}
 	for _, stack := range cfStacks.Stacks {
+		skip := true
 		for _, t := range stack.Tags {
 			if *t.Key == compose.ProjectTag {
-				status := compose.RUNNING
-				switch aws.StringValue(stack.StackStatus) {
-				case "CREATE_IN_PROGRESS":
-					status = compose.STARTING
-				case "DELETE_IN_PROGRESS":
-					status = compose.REMOVING
-				case "UPDATE_IN_PROGRESS":
-					status = compose.UPDATING
-				}
-				stacks = append(stacks, compose.Stack{
-					ID:     aws.StringValue(stack.StackId),
-					Name:   aws.StringValue(stack.StackName),
-					Status: status,
-				})
+				skip = false
 				break
 			}
 		}
+		if skip {
+			continue
+		}
+		status := compose.RUNNING
+		reason := ""
+		switch aws.StringValue(stack.StackStatus) {
+		case "CREATE_IN_PROGRESS":
+			status = compose.STARTING
+		case "DELETE_IN_PROGRESS":
+			status = compose.REMOVING
+		case "UPDATE_IN_PROGRESS":
+			status = compose.UPDATING
+		}
+		if status == compose.STARTING {
+			if err := s.CheckStackState(ctx, aws.StringValue(stack.StackName)); err != nil {
```

Comment:
from an architecture point of view, `sdk` should be limited to a thin wrapper on top of AWS go SDK, typically to make it easier with pointers and limit API scope to our own needs.

You could introduce `ListServiceTasks(stopped bool)` to return all tasks for a service (scoped to current deployment) and `GetTaskStoppedReason` to retrieve failure description, and invoke them from higher-level code, typically where we wait for stack completion: this is where you can get service ARN ("physical ID") from stack events, and then start per-service go routine to monitor service being successful to start Task(s).

Such simpler functions could later be reused to enhance `compose ps` command

Reply:
I refactored a bit, check it again pls.